### PR TITLE
Make sure to reset commit count in the cache on mirror syncing

### LINKED ISSUE #4715: Updated mirror shows old commit counter until gitea restart
- Gitea version (or commit ref): bc06ab4 
- Git version: not related
- Operating system: not related
- Database (use `[x]`): not related
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL) https://try.gitea.io/sageptr/mirror-test (until gitea update)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

Commit count is cached. It supposed to be invalidated after mirror update, but no.

How to reproduce:

1. Create new repository mirror.
2. Navigate this mirror content and look at commit count.
3. Push new commits to base repository.
4. Wait for mirror to synchronize.
5. Navigate to this mirror again. It will show new commits, but commit count will be old still.

### PR DESCRIPTION
**Pull request by adelowo**
Fixes #4715

### CODE REVIEW

**Review by lafriks — APPROVED** [review]

File: models/repo_mirror.go [hunk 1]
```diff
@@ -180,6 +181,8 @@ func (m *Mirror) runSync() bool {
 		}
 	}
 
+	cache.Remove(m.Repo.GetCommitsCountCacheKey(m.Repo.DefaultBranch, true))
```
Comment (SagePtr) [inline models/repo_mirror.go]: Does it reset default branch cache only? What about other branches cache?

**Review by SagePtr — COMMENTED** [review]
Does it reset default branch cache only? What about other branches cache?

File: models/repo_mirror.go (hunk 1 above)
Comment (adelowo) [reply models/repo_mirror.go]: Would fix

**Review by Bwko — APPROVED** [review]

**Comment by lunny** [comment]
@adelowo please send a back port PR to release/v1.5

**Comment by adelowo** [comment]
Done @lunny

**Pull request merged**